internal/vm/task.go

```diff
@@ -123,6 +123,13 @@ func (m *taskManager) newProc(taskID, execID string) (*vmProc, error) {
 	return proc, nil
 }
 
+func (m *taskManager) cleanupProc(taskID, execID string) {
+	_, err := m.deleteProc(taskID, execID)
+	if err != nil {
+		m.logger.WithField("TaskID", taskID).WithField("ExecID", execID).WithError(err).Error("failed to delete proc")
```

Comment:
I think the log line here should be more like "failed to cleanup proc after create or exec failure". It, in theory, should never return an error because it's only called after a successful call to `newProc`, so when it happens in practice it's a bug in some invariant we were trying to hold and should be distinguished from a normal failure to delete it.

Also, `m.logger` already includes the `TaskID` and `ExecID` fields, so we don't need to add them again.

Reply:
For things that shouldn't happen, can we just `panic`?

Reply:
We can, but I really prefer that we only panic if there is absolutely no other choice. While this shouldn't happen in theory, obviously there could be a bug now or in the future that causes it to arise. In that scenario, we absolutely should log it but I don't think this error specifically means we need to immediately crash everything. We can continue to operate.